tf2_ros/src/buffer.cpp

```diff
@@ -242,6 +242,7 @@ Buffer::waitForTransform(const std::string& target_frame, const std::string& sou
     // Immediately transformable
     geometry_msgs::msg::TransformStamped msg_stamped = lookupTransform(target_frame, source_frame, time);
     promise->set_value(msg_stamped);
+    callback(future);
```

Comment:
@ymd-stella I'd think you want this callback to be called in the `else if` clause below as well.

Reply:
I've fixed it.